tokio/src/sync/semaphore.rs

```diff
@@ -122,8 +122,38 @@ fn bounds() {
     check_send_sync_val(semaphore.acquire());
 }
 
+#[test]
+#[cfg(not(loom))]
+fn no_panic_at_maxpermits() {
+    let _ = Semaphore::new(Semaphore::MAX_PERMITS);
+    let s = Semaphore::new(Semaphore::MAX_PERMITS - 1);
+    s.add_permits(1);
+}
+
+#[test]
+#[cfg(not(loom))]
+#[should_panic]
+fn panic_when_exceeds_maxpermits1() {
+    let _ = Semaphore::new(Semaphore::MAX_PERMITS + 1);
+}
+#[test]
+#[cfg(not(loom))]
+#[should_panic]
+fn panic_when_exceeds_maxpermits2() {
+    let s = Semaphore::new(Semaphore::MAX_PERMITS - 1);
+    s.add_permits(1);
+    s.add_permits(1);
+}
```

Comment:
Tests go in `tests/sync_semaphore.rs`

Reply:
Moved the tests, also adjusted similar pre-existing tests.